config.go

```diff
@@ -32,7 +36,7 @@ func (o optionFunc) apply(c *config) {
 	o(c)
 }
 
-// Filter is a predicate used to determine whether a given http.request should
+// Filter is a predicate used to determine whether a given [http.Request] should
```

Comment:
I don't think this change is necessary.

Reply:
These are using the relatively new [doc linking](https://tip.golang.org/doc/comment#doclinks) format and make navigating from comments a bit more comfortable. 

But of course: if you'd prefer to keep this MR focused, I can just remove the changes.